opentracing-api-extensions-tracer/src/main/java/io/opentracing/contrib/api/tracer/APIExtensionsSpan.java

```diff
@@ -0,0 +1,282 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.api.tracer;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.api.SpanData;
+import io.opentracing.contrib.api.SpanObserver;
+
+public class APIExtensionsSpan implements Span, SpanData  {
+
+    private final Span wrappedSpan;
+
+    private String operationName;
+    private final long startTimestampMicro;
+    private long finishTimestampMicro;
+    private final long startTimeNano;
+    private long finishTimeNano;
+    private final Map<String,Object> tags;
+
+    private final List<SpanObserver> observers = new CopyOnWriteArrayList<SpanObserver>();
+
+    private final UUID correlationId = UUID.randomUUID();
+
+    /**
+     * This is the constructor for the extensions API span wrapper.
+     *
+     * @param span The span being wrapped
+     * @param operationName The operation name
+     * @param startTimestampMicro The start timestamp (microseconds)
+     * @param startTimeNano The start nano time, or 0 if the start timestamp was explicitly provided by the app
+     * @param tags The initial tags
+     */
+    public APIExtensionsSpan(Span span, String operationName,
+            long startTimestampMicro, long startTimeNano, Map<String,Object> tags) {
+        this.wrappedSpan = span;
```

Comment:
Would be nice to add some sanity checks here, like a null check for the `span`. Otherwise, it might fail later on with a NPE.